### DISCUSSION TITLE
[API Request] Save Play Queue using POST body

### DISCUSSION THREAD

**Discussion by kgarner7** [opensubsonic/open-subsonic-api#49] [API requests]
### Type of change

API extension

### Proposal description

Currently, the [savePlayQueue](http://www.subsonic.org/pages/api.jsp#savePlayQueue) endpoint is GET only (as are other endpoints). This is fine for small queues, but with play queues in the 1000s, you will run into 414 URI Too Long with reverse proxies. Having this endpoint also as POST will support a much larger queue (without hitting 414s).

### Backward compatibility impact

Since it would be a POST method, there should be no impact on the existing endpoint.

### Backward compatibility

- [X] No backward compatibility impact.

### API details

`POST` /rest/savePlayQueue
Parameters:
- `id`: song id in play queue, one per song (or a single field, comma separated)
- `current` (optional): id of current song
- `position` (optional, number): position in ms of current song

Data encoded as either `application/x-www-form-urlencoded` or `application/json`

### Security impacts

There should be sensible limits on the maximum size of the POST body (not sure what this number would be).

**Comment by Tolriq** [comment]
See https://github.com/opensubsonic/open-subsonic-api/discussions/26 the discussion was not started but the need is more global than just this endpoint.

There's the same issues with playlists and a few other cases.

We maybe should not use JSON for the XML users to not have to embed a new library just for that and limit to `application/x-www-form-urlencoded`

**Reply by kgarner7** [reply]
My bad! I didn't see that there was another issue for post as a whole (I just searched for "queue" and saw nothing). Would it be useful to list endpoints that could benefit from post body there?

**Reply by Tolriq** [reply]
I think globally all endpoints should support it no? `application/x-www-form-urlencoded` can easily just be used to pass the named args via the post as it would via a get. So it's a 1 to 1 match. Question being should server support both params via get and some via post.

**Reply by kgarner7** [reply]
I'm not sure if _all_ endpoints should be POST (at least, from the original meaning of POST being create?), but I don't really have a dog in this fight. Anyway, hopefully I didn't waste too much of your time...

**Reply by Tolriq** [reply]
Well no it's a good reminder to work on that one too :) Too many things to do. And it's not about should be post, it's about supporting post as a way to send parameters.
Since the API does not use the HTTP verbs for their meaning and everything is GET, supporting POST is just an alias for GET.
